source/mbed_trace.c

```diff
@@ -464,12 +484,18 @@ void mbed_vtracef(uint8_t dlevel, const char* grp, const char *fmt, va_list ap)
                 }
                 if (retval > 0) {
                     // not used anymore
-                    //ptr += retval;
-                    //bLeft -= retval;
+                    ptr += retval;
+                    bLeft -= retval;
                 }
             }
+            if (retval > 0 && bLeft > 0 && m_trace.fputs != mbed_trace_default_fputs) {
+                // add line feeds when using stdio:fputs streaming.
+                // by default we use custom fputs which behaviours same way than puts
+                // @todo this is a bit hack solution
+                retval = snprintf(ptr, bLeft, "\n");
+            }
             //print out whole data
-            m_trace.printf(m_trace.line);
+            m_trace.fputs(m_trace.line, m_trace.stream);
```

Comment:
previous `puts()` was used to print line. `fputs` behaviour a bit differently, but maybe we should use `fwrite` instead because we know line lenght already and it could be faster, opinions? 🤔